lib/puppet/provider/user/user_role_add.rb

```diff
@@ -26,8 +27,22 @@
     value !~ /\s/
   end
 
+  def shell=(value)
```

Comment:
Yeah, that is a somewhat questionable fallback and it does look like it should "just work"™ in the case of defining the flag.

The answer is apparently; it originally came from a customer and our notes say it was duplicated but don't give an example.  We've been carrying it since Puppet 3.x